src/components/magic/adeptPowers/PowerDetailRow.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {generateAddActionDetails, generateRemoveActionDetails} from './util/AdeptPowersUtil';
+import ModificationButton from '../../ModificationButton';
+import PowerLevelCounter from './PowerLevelCounter';
+
+class PowerDetailRow extends React.Component {
+	componentWillMount() {
+		this.setState({ value: '' });
+	}
+
+	getValue() {
+		return this.state.value;
+	}
+
+	updateValue(value) {
+		this.setState({ value });
+	}
+
+	powerBonus(bonus) {
+		if (!bonus) {
+			return 'N/A';
+		}
+		const bonusSelector = {
+			selectattribute: (attributes) => {
```

Comment:
Not a big deal, and don't feel like you have to, but there is a shorthand method definition for objects with es6. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions

So you don't need those arrows.

Assuming you copy/pasted this from me, it's because I didn't know about it at the time I made this.

Reply:
I learned with ES5, and just started touching ES6 a year ago (hence the arrow function).  I can go through and update it to be like so:
selectattribute(attributes) {... }